src/chrome/content/scaffold/scaffold.js

```diff
@@ -96,16 +96,21 @@ var Scaffold = new function() {
 		if(e.target !== document) return;
 		_document = document;
 
-		_browser = Components.classes["@mozilla.org/appshell/window-mediator;1"]
-						   .getService(Components.interfaces.nsIWindowMediator)
-						   .getMostRecentWindow("navigator:browser");
+		_browser = document.getElementsByTagName('browser')[0];
 
-		_browser.document.getElementById("content").tabContainer.addEventListener("TabSelect",
-			_updateFrames, true);
-		_browser.document.getElementById("appcontent").addEventListener("pageshow",
+		_browser.addEventListener("pageshow",
 			_updateFrames, true);
 		_updateFrames();
 		
+		let urlTextboxes = document.querySelectorAll('.browser-url');
+		for (let textbox of urlTextboxes) {
+			textbox.addEventListener('keyup', function(e) {
```

Comment:
Doesn't really matter, but `keypress` is probably a little more appropriate here.